docs/core/entity.md

```diff
@@ -119,6 +119,17 @@ The following properties are also available on entities. However, they are for a
 | state_attributes                | <code>dict &#124; None</code> | `None` | State attributes of a base domain. This property is implemented by the domain base entity and should not be implemented by integrations.
 | unit_of_measurement             | <code>str &#124; None</code> | The unit of measurement that the entity's state is expressed in. In most cases, for example for the `number` and `sensor` domains, this is implemented by the domain base entity and should not be implemented by integrations.
 
+## State attributes
+
+Entities can provide additional state attributes besides the built-in ones, to provide further details to its state.
+This should be done by providing a dictionary with keys and values to the `extra_state_attributes` property.
+
+Providing state attributes comes with the following rules:
+
+- If an attribute is expected, it should be in the dictionary. Attributes should not "come and go".
+- If an attribute is expected, but it's not providing a value right now, its value should be `None` and remain in the dictionary.
```

Comment:
I think the first 2 rules are quite similar. I also think we should explain what happens when for example the entity is unavailable. What should we do then?

Reply:
We could perhaps combine the first two as they talk about the same thing.
I'm not sure what you mean by `unavailable`? Should there be any special consideration for it?

Reply:
If an entity is unavailable, does it still read the extra state attributes property when writing state? And if so, should everything be set to None?

Reply:
State attributes are only set if the state of the entity is not unavailable. This is part of the base entity logic. Integrations should not adjust any logic concerning availability and state attributes.